s3/src/main/scala/org/apache/pekko/stream/connectors/s3/impl/S3Stream.scala

```diff
@@ -1024,13 +1024,14 @@ import scala.util.{ Failure, Success, Try }
   /**
    * Initiates a multipart upload. Returns a source of the initiated upload with upload part indicess
    */
-  private def initiateUpload(s3Location: S3Location,
+  private def initiateUpload(
+      s3Location: S3Location,
       contentType: ContentType,
       s3Headers: immutable.Seq[HttpHeader]): Source[(MultipartUpload, Int), NotUsed] =
     Source
       .single(s3Location)
       .flatMapConcat(initiateMultipartUpload(_, contentType, s3Headers))
-      .flatMapConcat(r => Source.repeat(r))
```

Comment:
What is the hypothetical problem with `Source.repeat` here?

Reply:
The flatmapConcat will need another submaterialization

Reply:
Ah so its a performance improvement, 👍